### PR TITLE
[libtorrent] minor version bump and python patch adjustment

### PR DESCRIPTION
**Pull request by FranciscoPombal**
Previously: https://github.com/microsoft/vcpkg/pull/12768

- Python patch was adjusted to apply once again. A recent minor change to the upstream files had made it incompatible.
- ~Include changes from recent commits to `RC_1_2`~ Bump to 1.2.9, which notably includes an important fix for the logic that selects the C++ version used during compilation.

Tested all features and they work fine.

### CODE REVIEW

**Comment by FranciscoPombal** [comment]
Please consider waiting a bit before merging this. 1.2.9 is going to be released sooner than I expected, so it's probably worth it to wait a bit and bump directly to the 1.2.9 tag instead. Apologies for the inconvenience.

File: ports/libtorrent/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: libtorrent
-Version: 1.2.8
+Version: 1.2.8-1
```
Comment (LilyWangL) [inline ports/libtorrent/CONTROL:2]: Can you use [Port-Version](https://github.com/microsoft/vcpkg/blob/b385e54baca496e07d31103d6b4731f79e0376ef/docs/maintainers/control-files.md#port-version) here?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/libtorrent/CONTROL (hunk 1 above)
Comment (FranciscoPombal) [reply ports/libtorrent/CONTROL:2]: @LilyWangL 

The docs say:

> It allows one to version the port file separately from the version of the underlying library; if you make a change to a port, without changing the underlying version of the library, you should increment this field by one

Technically, this is a version bump of the underlying library, not just a port change. Besides, I think we should actually wait for 1.2.9 (https://github.com/microsoft/vcpkg/pull/12946#issuecomment-674845423).

Comment (LilyWangL) [reply ports/libtorrent/CONTROL:2]: I will convert this PR to draft, waiting for this port update to version 1.2.9.

**Comment by FranciscoPombal** [comment]
@LilyWangL This is ready now.

**Review by LilyWangL — APPROVED** [review]

**Pull request merged**